inference-engine/src/cldnn_engine/CMakeLists.txt

```diff
@@ -22,19 +22,17 @@ ie_add_plugin(NAME ${TARGET_NAME}
               VERSION_DEFINES_FOR cldnn_engine.cpp)
 
 target_link_libraries(${TARGET_NAME} PRIVATE clDNN_lib pugixml
-                                             inference_engine inference_engine_legacy
-                                             inference_engine_transformations)
-
-if(USE_CNNNETWORK_LPT)
-    target_link_libraries(${TARGET_NAME} PRIVATE inference_engine_lp_transformations_legacy)
-else()
-    target_link_libraries(${TARGET_NAME} PRIVATE inference_engine_lp_transformations)
-endif()
+                                             inference_engine
+                                             inference_engine_transformations
+                                             inference_engine_lp_transformations)
 
 set(CLDNN_TOP_FOLDER "${IE_MAIN_SOURCE_DIR}/thirdparty/clDNN")
 target_include_directories(${TARGET_NAME} PRIVATE
         ${CMAKE_CURRENT_SOURCE_DIR}
+        $<TARGET_PROPERTY:inference_engine_plugin_api,INTERFACE_INCLUDE_DIRECTORIES>
         $<TARGET_PROPERTY:inference_engine_transformations,INTERFACE_INCLUDE_DIRECTORIES>
```

Comment:
I think we can also remove `inference_engine_transformations` and `ngraph` headers here since they come from `target_link_libraries( PRIVATE inference_engine_transformations ${NGRAPH_LIBRARIES})`. Just need to add `${NGRAPH_LIBRARIES}` to link libraries.

Reply:
Changed